Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/Systems/PerformanceAnalyticsSystem.cs

```diff
@@ -86,17 +80,19 @@ private void ReportPerformanceMetrics(float measureTime)
             jsonObjectBuilder.Set("player_count", 0); // TODO (Vit): How many users where nearby the current user
 
             // JS runtime memory
-            // bool isCurrentScene = scenesCache is { CurrentScene: { SceneStateProvider: { IsCurrent: true } } };
-            // JsMemorySizeInfo totalJsMemoryData = v8ActiveEngines.GetEnginesSumMemoryData();
-            // JsMemorySizeInfo currentSceneJsMemoryData = isCurrentScene ? v8ActiveEngines.GetEnginesMemoryDataForScene(scenesCache.CurrentScene.Info) : new JsMemorySizeInfo();
-            // jsonObjectBuilder.Set("jsheap_used", totalJsMemoryData.UsedHeapSizeMB);
-            // jsonObjectBuilder.Set("jsheap_total", totalJsMemoryData.TotalHeapSizeMB);
-            // jsonObjectBuilder.Set("jsheap_total_executable", totalJsMemoryData.TotalHeapSizeExecutableMB);
-            // jsonObjectBuilder.Set("jsheap_limit", totalJsMemoryData.HeapSizeLimitMB);
-            // jsonObjectBuilder.Set("jsheap_used_current_scene", !isCurrentScene ? -1f : currentSceneJsMemoryData.UsedHeapSizeMB);
-            // jsonObjectBuilder.Set("jsheap_total_current_scene", !isCurrentScene ? -1f : currentSceneJsMemoryData.TotalHeapSizeMB);
-            // jsonObjectBuilder.Set("jsheap_total_executable_current_scene", !isCurrentScene ? -1f : currentSceneJsMemoryData.TotalHeapSizeExecutableMB);
-            jsonObjectBuilder.Set("running_v8_engines", v8ActiveEngines.Count);
+            jsonObjectBuilder.Set("jsheap_used", DebugViewProfilingSystem.allScenesUsedHeapSize.ByteToMB());
+            jsonObjectBuilder.Set("jsheap_total", DebugViewProfilingSystem.allScenesTotalHeapSize.ByteToMB());
+            jsonObjectBuilder.Set("jsheap_total_executable", DebugViewProfilingSystem.allScenesTotalHeapSizeExecutable.ByteToMB());
+            jsonObjectBuilder.Set("jsheap_limit", DebugViewProfilingSystem.allScenesHeapSizeLimit.ByteToMB());
+
+            if (DebugViewProfilingSystem.currentSceneHasStats)
+            {
```

Comment:
calling one system from another system is a bad practice in ECS, since in theory Systems should be stateless (like functions). I know, we violate it a bit with having states on the system, but let's at least try to stick with no systems inter dependecies

Reply:
Normal practice to share/pass data between Systems is to use Components for it or to have some Service that injected in both. And that is why we have `Profiler.cs` class (which could be called `ProfilingService` but `Profiler` is just  shorter)